openlcsd/flow/task_wrapper.py

```diff
@@ -11,15 +11,13 @@ def after_return(self, status, retval, task_id, args, kwargs, einfo):
         """Handler called after the task returns."""
         if 'tmp_src_filepath' in args[0]:
             tmp_src_filepath = args[0].get('tmp_src_filepath')
-            if tmp_src_filepath:
-                tmp_dir = os.path.dirname(tmp_src_filepath)
-                if os.path.exists(tmp_dir):
-                    shutil.rmtree(tmp_dir, ignore_errors=True)
-
+            if tmp_src_filepath and os.path.exists(tmp_src_filepath):
+                os.remove(tmp_src_filepath)
         if 'src_dest_dir' in args[0]:
             src_dir = args[0].get('src_dest_dir')
             if src_dir and os.path.exists(src_dir):
-                shutil.rmtree(src_dir, ignore_errors=True)
+                if not args[0].get('sc_lable'):
```

Comment:
So for the forked task in the source container, after scanning, the source will not be removed?

Reply:
It will be removed after the sub-tasks finished the scan because the sub-tasks have no sc_lable in workflow but has the src_dir. This just keeps the source after finished fork_components_imports, which means the parent import is finished(This parent task workflow has the label, sc_lable).